db/db_sst_test.cc

```diff
@@ -629,6 +629,8 @@ TEST_F(DBSSTTest, DBWithMaxSpaceAllowed) {
   std::shared_ptr<SstFileManager> sst_file_manager(NewSstFileManager(env_));
   auto sfm = static_cast<SstFileManagerImpl*>(sst_file_manager.get());
 
+  SyncPoint::GetInstance()->DisableProcessing();
```

Comment:
You don't need this here, isn't it?

Reply:
Probably not. It's just a safe precaution. I wonder if we should, in the future, make all unit tests explicitly disable sync point, set up sync point callbacks/dependencies, and re-enable sync points. In the past, I have seen unit tests hitting strange memory access errors due to triggering a callback registered in another test and trying to access a local var that is already out-of-scope.

Reply:
But why? as of today, none of our unit tests which don't use sync-points disable them explicitly. 

> In the past, I have seen unit tests hitting strange memory access errors due to triggering a callback registered in another test and trying to access a local var that is already out-of-scope.

I guess this needs to be investigated deeper then. It feels out of place disable a sync point in a test which does not use it at all.

Reply:
The issue I ran into was that, a unit test register a lambda with a sync point but fail to clear the callbacks. Then another unit test runs, hits this sync point and calls the registered function.

Reply:
Giving it a second look, other unit tests in the same suite explicitly disable the sync point at the end. Therefore, you are right, it is better we remove it here.